Controllers/conversationController.js

```diff
@@ -0,0 +1,141 @@
+const CatchAsyncError = require('../Utils/CatchAsyncError')
+const db = require('../db/models/index')
+const User = db.User
+const UserConversation = db.UserConversation
+const Messages = db.Message
+const { Op } = require('sequelize')
+
+// WORK IN PROGRESS
+exports.createConversation = CatchAsyncError(async (req, res, next) => {
+  const searchedUserUsername = req.body.searchedUserUsername
+  const currentUserUsername = req.body.currentUserUsername
+
+  const searchedUserArr = await User.findAll({
+    where: {
+      username: searchedUserUsername
+    }
+  })
+  const currentUserArr = await User.findAll({
+    where: {
+      username: currentUserUsername
+    }
+  })
+  const currentUser = currentUserArr[0].dataValues
+  const searchedUser = searchedUserArr[0].dataValues
+
+  const currentUserConversations = await UserConversation.findAll({
+    attributes: ['ConversationId'],
+    where: {
+      UserId: currentUser.id
+    }
+  })
+  const containerCurrentUser = []
+
+  currentUserConversations.forEach(element => {
+    const [num] = (Object.values(element.dataValues))
+    containerCurrentUser.push(num)
+  })
+
+  const searchedUserConversations = await UserConversation.findAll({
+    attributes: ['ConversationId'],
+    where: {
+      UserId: searchedUser.id
+    }
+  })
+
+  searchedUserConversations.forEach(async element => {
+    const [num] = (Object.values(element.dataValues))
+    if (containerCurrentUser.includes(num)) {
+    //   const messages = []
+    //   const m = await Messages.findAll({
+    //     attributes: ['UserId', 'body'],
+    //     where: {
+    //       ConversationId: num
+    //     }
+    //   })
+    //   m.forEach(el => {
+    //     messages.push(el.dataValues)
+    //   })
+    //   res.status(200).json({
+    //     status: 'success',
+    //     messages
+    //   })
+      res.status(200).json({
+        status: 'success',
+        message: 'Chat exists'
+      })
+    }
+  })
```

Comment:
This seems unfinished.
In case the conversation already exists the response should return the existing conversation object. with a 200 response.
If the conversation does not already exist, create a new record, and return status 201.